helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -726,7 +726,7 @@ public boolean remove(String path, int options) {
       // case, we try to delete recursively
       _zkClient.delete(path);
     } catch (ZkException e) {
-      LOG.debug("Failed to delete {} with opts {}, err: {}. Try recursive delete", path, options,
+      LOG.info("Failed to delete {} with opts {}, err: {}. Try recursive delete", path, options,
```

Comment:
Just FYI, change from debug to info is a up-level. We may see more logs like this, is this what we want? (most default log level is at INFO or plus)

Reply:
Hmm you're right. What we're looking for is to diminish logging clutter - increasing the logging level would worsen that! I wanted to have the logging level consistent with the log error in the remove function but that wouldn't make sense to the overall objective of this fix. Thanks for the comment!

Reply:
The aim for this change is to make the log less misleading. By increasing the log level will indicate user that the previous warning log saying failed to remove znode won't be an issue and we will retry remove recursively. 
This is only in failure path and won't be causing log flooding.

Reply:
neat : I understand the overall intention of changing this to INFO level but I would be very cautious about this change since this is very core method which is used in many places. Any small changes in logging here would flood logs in specific scenarios. Also could you help me understand why was the "debug" log noise in the first place. It shouldn't even show up in logs unless debug logging is enabled right? (I assume debug logging is disabled by default)

Reply:
I am not too familiar with what the default logging is for the different applications of these methods. However, the ticket and issue created arose from customers (i.e. Venice) who have run into the issue of these "noisy" logs. Ideally, we would want to completely avoid calling these logs altogether (in the case that deletion succeeds further on). However, we weren't able to come up with a liable solution. Thus, if the log error is to be shown regardless, the purpose of this change as Xiaoyuan mentioned is to make it less misleading. Let me know what you think!

Reply:
I'm confused here.
If set the log level to info, then orignal code will NOT print debug log, but the new change will be printing more logs. It contradicts to the description.

> A log request of level p in a logger with level q is enabled if p >= q. This rule is at the heart of log4j. It assumes that levels are ordered. For the standard levels, we have ALL < DEBUG < INFO < WARN < ERROR < FATAL < OFF.
Here, in the original code p == debug and q == info. So the log won't be printed.

Debug to Info is considered upgrade instead of downgrade.

Reply:
You're right, I made a mistake in the description, thank you for pointing that out! I'll update the description with this feedback.